backend/Code4Ro.CoViz19.Reporter/Services/DataLoaderService.cs

```diff
@@ -0,0 +1,112 @@
+﻿using System;
+using System.Net.Http;
+using System.Security.Cryptography;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+using Code4Ro.CoViz19.Reporter.Extensions;
+using Code4Ro.CoViz19.Reporter.Options;
+using Code4Ro.CoViz19.Services;
+using Microsoft.Extensions.Caching.Memory;
+using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Options;
+using Newtonsoft.Json;
+
+namespace Code4Ro.CoViz19.Reporter.Services
+{
+    public class DataLoaderService : CronJobService
+    {
+        private const string CurrentDataHashKey = "current-data-hash";
+        private readonly IFileService _fileService;
+        private readonly ILogger<DataLoaderService> _logger;
+        private readonly PeanutOptions _peanutOptions;
+        private readonly HttpClient _peanutHttpClient;
+        private readonly IMemoryCache _cache;
+
+        public DataLoaderService(IScheduleConfig<DataLoaderService> config, IFileService fileService, IMemoryCache cache, IOptions<PeanutOptions> peanutOptions, ILogger<DataLoaderService> logger)
+            : base(config.CronExpression, config.TimeZoneInfo)
+        {
+            _fileService = fileService;
+            _logger = logger;
+            _cache = cache;
+            _peanutOptions = peanutOptions?.Value ?? throw new ArgumentNullException(nameof(peanutOptions));
+
+            _peanutHttpClient = new HttpClient();
+            _peanutHttpClient.DefaultRequestHeaders.Add(_peanutOptions.AuthHeader, _peanutOptions.AuthKey);
+        }
+
+        public override Task StartAsync(CancellationToken cancellationToken)
+        {
+            _logger.LogInformation($"{nameof(DataLoaderService)} starts.");
+            return base.StartAsync(cancellationToken);
+        }
+
+        public override async Task DoWork(CancellationToken cancellationToken)
+        {
+            try
+            {
+                _logger.LogInformation($"{DateTime.Now:hh:mm:ss} {nameof(DataLoaderService)} is working.");
+
+                var historicalPdfStats = await _fileService.GetRawData();
+                var currentHash = ComputeSha256Hash(historicalPdfStats);
+                var previousHash = _cache.Get<string>(CurrentDataHashKey);
+                if (currentHash != previousHash)
+                {
+                    _cache.Set(CurrentDataHashKey, currentHash);
+
+                    if (_peanutOptions.FeatureEnabled)
+                    {
+                        var response = await _peanutHttpClient.PostAsync(_peanutOptions.Url, new StringContent(@"{
+	                        ""outputs"":{
+                                ""3196x288"":""https://date-la-zi.ro/banners/ultraWide"",
+                                ""1280x720"":""https://date-la-zi.ro/banners/landscape"",
+                                ""1024x635"":""https://date-la-zi.ro/banners/landscape"",
+                                ""1080x1920"":""https://date-la-zi.ro/banners/portrait""
```

Comment:
I gave the wrong URLs here 😓 
```suggestion
                                ""3196x288"":""https://datelazi.ro/banners/ultraWide"",
                                ""1280x720"":""https://datelazi.ro/banners/landscape"",
                                ""1024x635"":""https://datelazi.ro/banners/landscape"",
                                ""1080x1920"":""https://datelazi.ro/banners/portrait""
```

Reply:
no problem i will fix this